redist/org.goldendict.desktop.desktop

```diff
@@ -5,5 +5,5 @@ Categories=Office;Dictionary;Education;Qt;
 Name=GoldenDict
 GenericName=Multiformat Dictionary
 Comment=GoldenDict
-Icon=goldendict
+Icon=org.goldendict.desktop
```

Comment:
Few of the image file names in _/usr/share/icons/hicolor/*/apps_ on my system follow this reverse-DNS scheme. There are practically no icon names like this except for *GNOME* and *Xfce* ones. Do you know why e.g. KDE icons don't follow this standard? Aren't very popular applications that don't conform to this standard (*firefox*, *libreoffice*, *qtcreator* to name a few) packaged on Flathub?

Reply:
I don't really manage firefox or so in flathub. But I think one of the reason is some flatpak doc doesn't quite follow the standard? @bilelmoussaoui should have more knowledge on this

Reply:
BTW some distros don't really follow the freedesktop standard strictly

Reply:
My Manjaro system *probably* preserves upstream file names as Arch does.

Reply:
Archlinux's packaging guidelines doesn't really mention the spec: https://www.linuxsecrets.com/archlinux-wiki/wiki.archlinux.org/index.php/Arch_packaging_standards.html

Yet if anyone realizes it and pushes it, it could be in the guidelines. But before that, there will be a transition.

Reply:
Since it preserves the upstream filename, it won't be a problem for transition

Reply:
Yes, I don't expect this to be a problem for most distros. But I still wonder why so few upstream apps follow this standard and whether GoldenDict should do so.

Reply:
Nobody does so because nobody really cared about those little details before.

Reply:
I just realized that renaming the icon will require updating icon themes that replace it. Since GoldenDict versions differ from distro to distro and many users manually build and install the latest version, I guess the icon themes will have to ship two copies of the icon - under the old and the new name. Is there any way to avoid this situation other than the obvious - not renaming the icon at all?

Reply:
The themes providing a symlink to the old icon?

Reply:
Yes, I guess the symlink can work. Though this is surely extra work to icon theme developers. Maybe that's the reason why so few icon names follow the reverse-DNS scheme - because of this renaming downside.

I just looked at this more carefully and couldn't find where the reverse-DNS scheme is prescribed for the icon names. Is the icon file naming actually specified somewhere? If yes, then where?

Reply:
I don't think it's part of the icon theme specs, no. But it does make sense for the exact same reason that your desktop file, metainfo, dbus services or anything app specific to be prefixed with app-id.

Reply:
It _might_ make sense. But this is not sufficient ground for renaming and the icon theme symlinks in my opinion. GoldenDict is based on Qt, so it makes more sense to follow KDE's practice than GNOME's or Xfce's. Therefore I don't want this pull request to rename the icon.

Reply:
>  It might make sense. But this is not sufficient ground for renaming and the icon theme symlinks in my opinion. 

Well, there has to be a transition when changes are coming... And the easiest way to do the migration is to strat from the upstream.

So at least, apart from the icon name, kde software uses the reversed DNS name already (e.g. https://invent.kde.org/utilities/kate/-/blob/master/kate/data/org.kde.kate.desktop). And I have to change the icon name at least in the build procedure to publish it into flathub.

Reply:
> Well, there has to be a transition when changes are coming... And the easiest way to do the migration is to strat from the upstream.

Why do you believe that icon names are about to be standardized?

From your link: `Icon=kate`. The same icon naming scheme is used by all other KDE applications installed in my system. I don't object to the renaming of *goldendict.desktop*, *goldendict.appdata.xml* or the discussed change of the `<id>org.goldendict.desktop</id>` line, because these indeed don't conform to published specifications. I also don't see any harm in these spec-conformance changes. But once you update this pull request, I'll ask the opinion of those who contributed to *goldendict.appdata.xml* in the past - to be sure that all packagers agree.

> And I have to change the icon name at least in the build procedure to publish it into flathub.

Since the icon naming is not standardized, this looks like a bug in Flathub.

Reply:
In this part of the flathub doc it says so: https://docs.flatpak.org/en/latest/conventions.html#application-icons

> Desktop files should be named with the application’s ID, followed by the .desktop file extension, and should be placed in /app/share/applications/. 

And yes in https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#file_formats it doesn't say the icon filename should be the same as the application id. But it sounds more ideal that all the ids are the same. Otherwise it might be a bit confusing. But since it's not standardized, I can't provide any proof that it was a mistake or they forgot to update the icon theme spec, since I'm not the one who wrote the specs (neither the flatpak spec or the freedesktop specs).

Reply:
I'll wait for @bilelmoussaoui if there are more to elaborate. If no, I'll just rollback that part and keep the other.

Reply:
@vedgy The desktop files are supposed to be using reverse DNS as described [in the specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#file-naming) and as you know, very few apps actually follow it. I think the effort on setting fairly arbitrary specifications without actual enforcement lead to the reality of various desktop specifications not being updated at all. The author of flatpak is also the co-author of the FreeDesktop icons specification, so I'd say it's not a stretch to assume it could have been added here (and then ignored anyway by everyone but flatpak packagers).

Obviously if it's something you don't agree with, it is what it is. I don't think there's any point in forcing anyone to rename something.

Reply:
Some themes includes their own icon for the GoldenDict and after renaming `goldendict.png` to `org.goldendict.desktop` you will break them.

Reply:
Please refer to the latest changes. Not true now